app/models/service_template_ansible_playbook.rb

```diff
@@ -24,13 +24,41 @@ def self.default_retirement_entry_point
   #     :reconfigure (same as provision)
   #
   def self.create_catalog_item(options, _auth_user)
+    task_id = create_catalog_item_queue(options, _auth_user)
+    task = MiqTask.wait_for_taskid(task_id)
+    task.task_results
```

Comment:
@mkanoor - I left this for now - I'm not entirely sure how you wanted to handle just having the `id` for the `task_results` in MiqTask.

Also - regarding: `raise MiqException::Error, task.message unless task.status == "Ok"` Were you thinking to raise the exception before the task times out? Currently the job will fail and the task is recording why without the exception.